miio/integrations/vacuum/dreame/dreamevacuum_miot.py

```diff
@@ -190,37 +274,132 @@ def volume(self) -> str:
+
+    @property
+    def cleaning_time(self) -> str:
+        return self.data["cleaning_time"]
+
+    @property
+    def cleaning_area(self) -> str:
+        return self.data["cleaning_area"]
 
-class DreameVacuumMiot(MiotDevice):
-    """Interface for Vacuum 1C STYTJ01ZHM (dreame.vacuum.mc1808)"""
+    @property
+    def first_clean_time(self) -> str:
+        return self.data["first_clean_time"]
 
-    mapping = _MAPPING
-    _supported_models = ["dreame.vacuum.mc1808"]
+    @property
+    def total_clean_time(self) -> str:
+        return self.data["total_clean_time"]
+
+    @property
+    def total_clean_times(self) -> str:
+        return self.data["total_clean_times"]
+
+    @property
+    def total_clean_area(self) -> str:
+        return self.data["total_clean_area"]
+
+    @property
+    def cleaning_mode(self):
+        cleaning_mode = self.data["cleaning_mode"]
+        if self.model == DREAME_1C:
+            try:
+                return CleaningModeDreame1C(cleaning_mode)
+            except ValueError:
+                _LOGGER.error(f"Unknown CleaningMode ({cleaning_mode})")
+                return CleaningModeDreame1C.Unknown
+        elif self.model in [DREAME_D9, DREAME_F9]:
+            try:
+                return CleaningModeDreameF9(cleaning_mode)
+            except ValueError:
+                _LOGGER.error(f"Unknown CleaningMode ({cleaning_mode})")
+                return CleaningModeDreameF9.Unknown
+
+    @property
+    def life_sieve(self) -> Optional[str]:
+        return self.data.get("life_sieve")
+
+    @property
+    def life_brush_side(self) -> Optional[str]:
+        return self.data.get("life_brush_side")
+
+    @property
+    def life_brush_main(self) -> Optional[str]:
+        return self.data.get("life_brush_main")
+
+    # TODO: get/set water flow for Dreame 1C
+    @property
+    def water_flow(self) -> Optional[WaterFlow]:
+        try:
+            water_flow = self.data["water_flow"]
+        except KeyError:
+            return None
+        try:
+            return WaterFlow(water_flow)
+        except ValueError:
+            _LOGGER.error("Unknown WaterFlow (%s)", self.data["water_flow"])
+            return WaterFlow.Unknown
+
+    @property
+    def is_water_box_carriage_attached(self) -> Optional[bool]:
+        """Return True if water box carriage (mop) is installed, None if sensor not
+        present."""
+        if "water_box_carriage_status" in self.data:
+            return self.data["water_box_carriage_status"] == 1
+        return None
+
+
+class DreameVacuum(MiotDevice):
+    _supported_models = [
+        DREAME_1C,
+        DREAME_D9,
+        DREAME_F9,
+    ]
+    _mappings = MIOT_MAPPING
+
+    def get_fanspeeds(self):
+        """Return fanspeeds enum for model if found or None."""
+        if self.model == DREAME_1C:
+            return CleaningModeDreame1C
+        elif self.model in [DREAME_F9, DREAME_D9]:
+            return CleaningModeDreameF9
```

Comment:
```suggestion
    def fan_speed_presets(self) -> Dict[str, int]:
        """Return fanspeeds enum for model if found or None.""" 
        if self.model == DREAME_1C:
            return _enum_as_dict(CleaningModeDreame1C)
        elif self.model in [DREAME_F9, DREAME_D9]:
            return _enum_as_dict(CleaningModeDreameF9)
            
        _LOGGER.warning("Unknown model, falling back to %r", CleaningModeDreame1C)
        return _enum_as_dict(CleaningModeDreame1C)
```
I'm not sure what should be used here as a fallback, but renaming this method (as well as the `set_` one) will make it compatible with the roborock implementation that will simplify generalizing these.